packages/core/src/indexing-store/index.ts

```diff
@@ -38,17 +39,28 @@ export const parseSqlError = (e: any): Error => {
   return error;
 };
 
-export const validateUpdateSet = (table: Table, set: Object): Object => {
+export const validateUpdateSet = (
+  table: Table,
+  set: Object,
+  prev: Row | null,
+): Object => {
   const primaryKeys = getPrimaryKeyColumns(table);
+  const newSet = { ...set } as Row;
 
   for (const { js } of primaryKeys) {
     if (js in set) {
-      throw new NonRetryableError(
-        `Primary key column '${js}' cannot be updated`,
-      );
+      // Note: Strip off the primary keys if they are identical, otherwise throw an error.
+      if (prev !== null && newSet[js] === prev[js]) {
+        delete newSet[js];
+      } else {
```

Comment:
Is it necessary to delete from `newSet` when the pks are equal? It would save memory if we didn't do this.